pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -106,6 +109,19 @@ private CompletableFuture<Void> sendTopicPolicyEvent(TopicName topicName, Action
                     new BrokerServiceException.NotAllowedException("Not allowed to send event to health check topic"));
         }
         CompletableFuture<Void> result = new CompletableFuture<>();
+        try {
+            final Optional<Policies> namespacePolicies = pulsarService.getPulsarResources().getNamespaceResources()
```

Comment:
Could you use the asynchronous method?

Reply:
done. However I had to move all the method to be async, I'm not fully happy because I thought we could cherry-pick to 210 and 211 easily and keeping it as a small change, but here we are